### PR TITLE
simplified aws s3 docs

### PR DESCRIPTION
**Pull request by OskarStark**
refs #191 

Now all versions are in one file and a doc for every version is available.

Another possibility would be to split it into 3 files, but name the sdk version in the filename

### CODE REVIEW

File: Resources/docs/adapters/awss3.md [hunk 1]
```diff
@@ -63,3 +110,52 @@ knp_gaufrette:
...
+ * `amazon_s3_id`: the id of the AmazonS3 service used for the underlying connection
+ * `bucket_name`: the name of the bucket to use
+ * `options`: additional (optional) settings
+   * `directory`: the directory to use, within the specified bucket
+   * `region`
+   * `create`
+
+### Example
+
+Once the service is set up use its key as the amazon_s3_id` in the gaufrette configuration:
```
Comment (nicolasmure) [inline Resources/docs/adapters/awss3.md]: looks like there's a missing backquote on this line

File: Resources/docs/adapters/awss3.md [hunk 2]
```diff
@@ -63,3 +110,52 @@ knp_gaufrette:
...
+    adapters:
+        foo:
+            amazon_s3:
+                amazon_s3_id: amazonS3
+                bucket_name: foo_bucket
+                options:
+                    directory: foo_directory
+```
+
+Not`e that the SDK seems to have some issues with bucket names with dots in them, e.g. "com.mycompany.bucket" seems to have issues but "com-mycompany-bucket" works.
```
Comment (nicolasmure) [inline Resources/docs/adapters/awss3.md]: can you remove this backquote please ?

File: Resources/docs/adapters/awss3.md (hunk 1 above)
Comment (OskarStark) [reply Resources/docs/adapters/awss3.md]: fixed, thx

File: Resources/docs/adapters/awss3.md (hunk 2 above)
Comment (OskarStark) [reply Resources/docs/adapters/awss3.md]: fixed, thx

**Comment by OskarStark** [comment]
> Thank you :) . Can you add a simple table of content in the beginning of the doc file (referencing the v3, v2 and v1 sections), and also update the README.md file to remove the link to the deleted amazon_s3.md file please ?

sure thing, shall we name it "AWS" or "Amazon S3" or "Amazon S3 (AWS)" for the filename link, headline etc.?

**Comment by nicolasmure** [comment]
I think it would be nice to name it `AwsS3`, to keep it coherent with the [`Gaufrette\Adapter\AwsS3`](https://github.com/KnpLabs/Gaufrette/blob/dd0d2e9ef88ff1deb50881ed8de2e1d07617817a/src/Gaufrette/Adapter/AwsS3.php) adapter (which is used with the maintained amazon sdk).

**Comment by OskarStark** [comment]
done!

File: Resources/docs/adapters/awss3.md [hunk 3]
```diff
@@ -1,8 +1,33 @@
-# Amazon S3 SDK v2
+# AWS S3
 
-Adapter for Amazon S3 SDK v2.
```
Comment (nicolasmure) [inline Resources/docs/adapters/awss3.md]: Can you add a table of content referencing the three sections of this file please ?
```markdown
- [Amazon SDK v3 (recommended)](#adapter-for-amazon-s3-sdk-v3-recommended)
- [Amazon SDK v2](#adapter-for-amazon-s3-sdk-v2)
- [Amazon SDK v1 (deprecated)](#adapter-for-amazon-s3-sdk-v1-deprecated-by-amazon)
```

File: Resources/docs/adapters/awss3.md [hunk 4]
```diff
@@ -13,7 +44,33 @@ Adapter for Amazon S3 SDK v2.
    This directory will be created in the root of the bucket and all files will be read and written there.
    * `acl` Default ACL to apply to the objects
 
-## Defining services
+Also note that when you create a bucket it [is located in a specific region](http://docs.aws.amazon.com/AmazonS3/latest/dev/UsingBucket.html).
```
Comment (nicolasmure) [inline Resources/docs/adapters/awss3.md]: Can you remove this line and the one below please (they're already [here](https://github.com/KnpLabs/KnpGaufretteBundle/pull/193/files#diff-bd830abb396dedb7f711473d5371dd96R33)). After that it'll be OK to merge ;)

Comment (OskarStark) [reply Resources/docs/adapters/awss3.md]: done in a separate commit, are you fine with it?
I could squash my commits if you want

Comment (nicolasmure) [reply Resources/docs/adapters/awss3.md]: Yes it would be nice if you could squash them :)

Comment (OskarStark) [reply Resources/docs/adapters/awss3.md]: done :)

**Comment by nicolasmure** [comment]
Thank you @OskarStark !

**Comment by OskarStark** [comment]
You are welcome 👍 😄

**Pull request merged**